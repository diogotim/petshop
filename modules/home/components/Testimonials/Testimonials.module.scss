@import '../../../../common/styles/mixins';

.root {
    width: 100%;
    margin-top: 4rem;
    display: flex;
    flex-wrap: wrap;
    align-items:flex-start;
}

.testimonialProfile {
    display: none;
}

@include md {
    .testimonialProfile {
        display: block;
        position: relative;
        width: 30%;
        height: 40vh;
        overflow: hidden;
        border-radius: 4px;
    }
}

.speechWrapper {
    flex: 1;
}

.clientName {
    text-align: right;
}

.speech {
    display: block;
    font-family: var(--display-font);
    position: relative;
    padding: 2rem;
    font-size: 1.2em;
    color: #fff;
    border-radius: 8px;
    background: var(--main);
}


@include md {
    .speech {
        margin-left: 2rem;
    }

    .speech::before {
        content: "";
        width: 0px;
        height: 0px;
        position: absolute;
        left: -4rem;
        top: 4rem;
        border-left: 2rem solid transparent;
        border-top: 2rem solid var(--main);
        border-right: 2rem solid var(--main);
        border-bottom: 2rem solid transparent;
        z-index: 9;
    }
}