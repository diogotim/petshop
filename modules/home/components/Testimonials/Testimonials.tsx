import React from "react";
import Link from "next/link";
import classes from "./Testimonials.module.scss";

const Testimonials: React.FunctionComponent = () => {
  return (
    <div className="container">
      <div className={classes.root}>
        <div className={`${classes.testimonialProfile} shadow-md`}>
          <img
            className={`imgCover imgFill`}
            src={require("../../../../public/woman-testimonials.jpg")}
          />
        </div>
        <div className={classes.speechWrapper}>
          <p className={`${classes.speech} shadow-md`}>
            “Etiam facilisis ornare ligula, id feugiat ante laoreet eget. Class
            aptent taciti sociosqu ad litora torquent per conubia nostra, per
            inceptos himenaeos. Etiam sed dui et mauris faucibus suscipit. Duis
            maximus purus id tincidunt dictum. Sed feugiat diam ut enim
            sollicitudin vestibulum.”
          </p>
          <p className={classes.clientName}>— Someone</p>
        </div>
      </div>
    </div>
  );
};

export default Testimonials;
