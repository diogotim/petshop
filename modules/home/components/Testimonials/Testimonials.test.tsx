import { render } from "@testing-library/react";
import Testimonials from "./Testimonials";

describe("Testimonials", () => {
  it("renders the status card unchanged", () => {
    const testimonials = render(<Testimonials />);
    expect(testimonials).toMatchSnapshot();
  });
});
