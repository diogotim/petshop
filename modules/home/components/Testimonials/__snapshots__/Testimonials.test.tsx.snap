// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testimonials renders the status card unchanged 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <div
          class="root"
        >
          <div
            class="testimonialProfile shadow-md"
          >
            <img
              class="imgCover imgFill"
              src="test-file-stub"
            />
          </div>
          <div
            class="speechWrapper"
          >
            <p
              class="speech shadow-md"
            >
              “Etiam facilisis ornare ligula, id feugiat ante laoreet eget. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Etiam sed dui et mauris faucibus suscipit. Duis maximus purus id tincidunt dictum. Sed feugiat diam ut enim sollicitudin vestibulum.”
            </p>
            <p
              class="clientName"
            >
              — Someone
            </p>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <div
        class="root"
      >
        <div
          class="testimonialProfile shadow-md"
        >
          <img
            class="imgCover imgFill"
            src="test-file-stub"
          />
        </div>
        <div
          class="speechWrapper"
        >
          <p
            class="speech shadow-md"
          >
            “Etiam facilisis ornare ligula, id feugiat ante laoreet eget. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Etiam sed dui et mauris faucibus suscipit. Duis maximus purus id tincidunt dictum. Sed feugiat diam ut enim sollicitudin vestibulum.”
          </p>
          <p
            class="clientName"
          >
            — Someone
          </p>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
